BigriverBookstore-apple/Sources/Controller/LibraryBooksController.swift

```diff
@@ -8,12 +8,43 @@
 
 import Cocoa
 
-class LibraryBooksController: NSViewController {
-    var libraryBooks = LibraryBooksViewModel()
-
+class LibraryBooksController: NSViewController, NSCollectionViewDataSource {
+    @IBOutlet var libraryBooksCollectionView: NSCollectionView!
+    private var libraryBooks = LibraryBooksViewModel()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         libraryBooks.loadBookList()
-        libraryBooks.refreshData = {}
+        libraryBooks.refreshData = {
+            self.libraryBooksCollectionView.reloadData()
```

Comment:
@c-Viorel I think the reloadData is not on the main thread. We should make sure we call UI methods only on the main thread.

Reply:
I really don't have an explanation for it. I don't have any idea how I'd missed it.